coding_systems/base/import_data_utils.py

```diff
@@ -136,3 +149,118 @@ def setup_new_import_database(self):
         call_command("migrate", self.coding_system, database=database_alias)
 
         return database_alias
+
+
+def update_codelist_version_compatibility(coding_system_id, new_database_alias):
+    new_coding_system = CODING_SYSTEMS[coding_system_id](
+        database_alias=new_database_alias
+    )
+    """
+    Check compatibility of existing codelist versions with a new coding system release
+
+    Filter codelist versions to only those that are compatible with, or created from,
+    the previous release. Any codelist versions that were not compatible with the previous
+    release can ignored, as we wouldn't expect them to be compatible with any later releases.
+
+    Note we don't exlude version created from or already in the compatible
+    set for THIS release.  If we're doing a forced-overwrite import (which is where
+    we'd expect that might happen), we want to re-check compatibility
+
+    Then check
+    1) versions with searches by rerunning the searches with the new release and checking
+       the results are the same
+    2) versions without searches, but with hierarchies, by recreating the hierarchy with the
+       new release and checking if it is identical to the version's cached hierarchy
+    """
+    previous_release = (
+        CodingSystemRelease.objects.filter(coding_system=coding_system_id)
+        .exclude(id=new_coding_system.release.id)
+        .latest("valid_from")
+    )
```

Comment:
I'm quite paranoid...  I feel it would be helpful to assert here that `previous_release.valid_from` is earlier than new_coding_system.release.valid_from`.

Reply:
hmmm...it should be.  Let me check if I've forgotten to push something

Reply:
I've added that now - I had updated it so it filtered by valid_from < new_coding_system.release.valid_from, but that seems to have gone missing in a rebase somewhere